src/shmem_team.c

```diff
@@ -0,0 +1,455 @@
+            ret = shmem_internal_team_split_strided(parent_team, start, parent_stride,
+                                            xsize, xaxis_config, xaxis_mask, xaxis_team);
+            if (ret) {
+                RAISE_ERROR_STR("x-axis 2D strided split failed");
+            }
+        }
+        start += xrange * parent_stride;
+    }
+
+    start = parent_start;
+
+    for (int i = 0; i < num_yteams; i++) {
+        int remainder = parent_size % xrange;
+        int yrange = parent_size / xrange;
+        int ysize = (remainder && i < remainder) ? yrange + 1 : yrange;
+
+        if (shmem_internal_pe_in_active_set(shmem_internal_my_pe, start, xrange*parent_stride, ysize) != -1) {
+            ret = shmem_internal_team_split_strided(parent_team, start, xrange*parent_stride,
+                                            ysize, yaxis_config, yaxis_mask, yaxis_team);
+            if (ret) {
+                RAISE_ERROR_STR("y-axis 2D strided split failed");
+            }
+        }
+        start += parent_stride;
+    }
+
+    size_t psync = shmem_internal_team_choose_psync(parent_team, SYNC);
+
+    shmem_internal_barrier(parent_start, parent_stride, parent_size,
+                           &shmem_internal_psync_barrier_pool[psync]);
+
+    shmem_internal_team_release_psyncs(parent_team, SYNC);
+
+    return 0;
+}
+
+int shmem_internal_team_destroy(shmem_internal_team_t *team)
+{
+
+    if (team == SHMEMX_TEAM_INVALID || team == &shmem_internal_team_world ||
+        team == &shmem_internal_team_shared) {
+        return -1;
+    } else if ((psync_pool_avail[team->psync_idx / CHAR_BIT] >> team->psync_idx) & 1) {
+        RAISE_WARN_STR("Destroying a team without an active pSync");
+    } else {
+        for (size_t i = 0; i < PSYNC_CHUNK_SIZE; i++) {
+            shmem_internal_psync_pool[team->psync_idx * PSYNC_CHUNK_SIZE+ i] = SHMEM_SYNC_VALUE;
+        }
+        for (size_t i = 0; i < SHMEM_SYNC_SIZE; i++) {
+            shmem_internal_psync_barrier_pool[team->psync_idx * SHMEM_SYNC_SIZE + i] = SHMEM_SYNC_VALUE;
+        }
+        shmem_internal_bit_set(psync_pool_avail, N_PSYNC_BYTES, team->psync_idx);
+    }
+
+    /* Destroy all undestroyed shareable contexts on this team */
+    for (size_t i = 0; i < team->contexts_len; i++) {
+        if (team->contexts[i] != NULL) {
+            if (team->contexts[i]->options & SHMEM_CTX_PRIVATE)
+                RAISE_WARN_MSG("Shutting down with unfreed private context (%zu)\n", i);
+            shmem_transport_quiet(team->contexts[i]);
+            shmem_transport_ctx_destroy(shmem_internal_team_world.contexts[i]);
+        }
+    }
+    free(team->contexts);
+    free(team);
+    shmem_internal_team_pool[team->psync_idx] = NULL;
+
+    return 0;
+}
+
+int shmem_internal_team_create_ctx(shmem_internal_team_t *team, long options, shmem_ctx_t *ctx)
+{
+    int ret = shmem_transport_ctx_create(team, options, (shmem_transport_ctx_t **) ctx);
+    SHMEM_ERR_CHECK_NULL(ctx, 0);
```

Comment:
The argument checks should not be used outside of the public bindings, but also the second argument will cause this to always be a nop.  Just delete this line.